src/cmake/build_libuhdr.cmake

```diff
@@ -20,6 +20,18 @@ if (TARGET libjpeg-turbo::jpeg)
     get_target_property(JPEG_LIBRARY JPEG::JPEG INTERFACE_LINK_LIBRARIES)
 endif ()
 
+if (DEFINED ENV{UHDR_C})
+    set (UHDR_CMAKE_C_COMPILER $ENV{UHDR_C})
+else ()
+    set (UHDR_CMAKE_C_COMPILER ${CMAKE_C_COMPILER})
+endif()
+
+if (DEFINED ENV{UHDR_CXX})
+    set (UHDR_CMAKE_CXX_COMPILER $ENV{UHDR_CXX})
+else ()
+    set (UHDR_CMAKE_CXX_COMPILER ${CMAKE_CXX_COMPILER})
+endif()
+
```

Comment:
I think this will be simpler, using our `set_cache` utility that we can give a default but allow automatic override by matching env variable:

```
set_cache (UHDR_CMAKE_CXX_COMPILER ${CMAKE_CXX_COMPILER} "libuhdr build compiler override" ADVANCED)
set_cache (UHDR_CMAKE_C_COMPILER ${CMAKE_C_COMPILER} "libuhdr build compiler override" ADVANCED)
```